examples/data-sources/cockroach_restores/data-source.tf

```
@@ -0,0 +1,20 @@
+# Fetch the latest restore job
+data "cockroach_restores" "latest_restore" {
+  cluster_id = cockroach_cluster.prod_cluster.id
+  sort_order = "DESC"
+  limit      = 1
+}
+
+# Get the id of the most recent restore job
+output "latest_restore_id" {
+  value = data.cockroach_restores.latest_restore.restores[0].id
+}
+
+# Fetch restore jobs within a specific date range
+data "cockroach_restores" "restores_in_range" {
+  cluster_id = cockroach_cluster.prod_cluster.id
+  start_time = "2025-08-15"
+  end_time   = "2025-08-19"
```

nit:  I'm thinking it would help people understand the exclusivity if we added a comment here  @jhlodin are you good with that?
```suggestion
  # get restores up to but not including 8/19
  end_time   = "2025-08-19"
```

---

added this comment but @jhlodin lmk if you have any objections